src/NavigationMap.ts

```diff
@@ -65,8 +65,36 @@ export default class NavigationMap {
   }
 
   public getTokenForRange(range: Range) {
-    return Object.values(this.map).find(
+    const matches = Object.values(this.map).filter(
       (token) => token.range.intersection(range) != null
     );
+
+    // If multiple matches sort and take the first
```

Comment:
I would expand to containing tokens, not just take the first.  Eg if anchor and active fall in the middle of tokens, return beginning / end of anchor to beginning / end of active, depending whether reversed

Reply:
This has nothing to do with range it's just if the cursor is between two tokens then we have to select one.

Reply:
Yes but if your selection spans across two tokens, "take token" should expand to the containing tokens.  This code will cause it to select only the first token